pkg/dns/aws/dns.go

```diff
@@ -75,44 +88,74 @@ type Config struct {
+// openshift.io/ingress-operator/operatorReleaseVersion to the user-agent
+// request header.
 func NewProvider(config Config, operatorReleaseVersion string) (*Provider, error) {
+	caWatcher, err := watcher.New(defaultCABundle)
+	if err != nil {
+		return nil, err
+	}
+
+	sess, err := NewProviderSession(config, operatorReleaseVersion, caWatcher.GetFileData())
+	if err != nil {
+		return nil, err
+	}
+
+	provider := &Provider{
+		elb:         elb.New(sess, aws.NewConfig().WithRegion(config.Region)),
+		route53:     route53.New(sess),
+		fileWatcher: caWatcher,
+		// TODO: This API will only return hostedzone resources (which are global)
+		// when the region is forced to us-east-1. We don't yet understand why.
+		tags:           resourcegroupstaggingapi.New(sess, aws.NewConfig().WithRegion("us-east-1")),
+		config:         config,
+		idsToTags:      map[string]map[string]string{},
+		lbZones:        map[string]string{},
+		updatedRecords: sets.NewString(),
+	}
+	return provider, nil
+}
+
+// NewProviderSession returns a new AWS Session with credentials from config;
+// adding openshift.io/ingress-operator/operatorReleaseVersion to the user-agent
+// request header and pemData to the TLS Config CA pool.
+func NewProviderSession(config Config, operatorReleaseVersion string, pemData []byte) (*session.Session, error) {
 	creds := credentials.NewStaticCredentials(config.AccessID, config.AccessKey, "")
+	caPool := x509.NewCertPool()
+	if ok := caPool.AppendCertsFromPEM(pemData); !ok {
+		return nil, fmt.Errorf("failed to append certificates to ca pool")
+	}
+	tlsConfig := &tls.Config{RootCAs: caPool}
+	trans := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		TLSClientConfig:       tlsConfig,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
 	sess, err := session.NewSessionWithOptions(session.Options{
 		Config: aws.Config{
 			Credentials: creds,
+			HTTPClient:  &http.Client{Transport: trans},
 		},
 		SharedConfigState: session.SharedConfigEnable,
 	})
 	if err != nil {
-		return nil, fmt.Errorf("couldn't create AWS client session: %v", err)
+		return nil, fmt.Errorf("failed to create AWS client session: %v", err)
 	}
 	sess.Handlers.Build.PushBackNamed(request.NamedHandler{
 		Name: "openshift.io/ingress-operator",
 		Fn:   request.MakeAddToUserAgentHandler("openshift.io ingress-operator", operatorReleaseVersion),
 	})
 
-	region := aws.StringValue(sess.Config.Region)
-	if len(region) > 0 {
-		log.Info("using region from shared config", "region name", region)
-	} else {
-		region = config.Region
-		log.Info("using region from operator config", "region name", region)
-	}
-	if len(region) == 0 {
-		return nil, fmt.Errorf("region is required")
-	}
```

Comment:
I'm not sure it is safe to move this validation from the session setup logic here to `createDNSProvider`.  Here, we validate the region from the session configuration, which may be reading from `~/.aws/config` or may be using the region from the cluster config, but `createDNSProvider` only has the cluster config, so it is not validating the same thing.  @ironcladlou, do I have that right?  It should only matter for local development though, so maybe it isn't worth the trouble to check the session config.

Reply:
@Miciah see https://github.com/openshift/cluster-ingress-operator/pull/334#discussion_r355739759, the check is bubbled up to the aws provider of  `createDNSProvider()`.